src/frontend/apps/lti_site/components/VideoWidgetProvider/widgets/Transcripts/index.tsx

```diff
@@ -0,0 +1,185 @@
+
+const messages = defineMessages({
+  info: {
+    defaultMessage:
+      'This widget allows you to read and downloads transcripts for the video.',
+    description: 'Info of the widget used for uploading transcripts.',
+    id: 'components.UploadTranscripts.info',
+  },
+  title: {
+    defaultMessage: 'Transcripts',
+    description: 'Title of the widget used for read and download transcripts.',
+    id: 'components.UploadTranscripts.title',
+  },
+  hideTranscript: {
+    defaultMessage: 'Hide transcript',
+    description: 'Text to hide a displayed transcript',
+    id: 'components.Transcripts.hideTranscript',
+  },
+  transcriptSelectPlaceholder: {
+    defaultMessage: 'Choose a language',
+    description: 'Placeholder for the transcript select box',
+    id: 'components.Transcripts.transcriptSelect',
+  },
+  transcriptDownload: {
+    defaultMessage: 'Download',
+    description: 'Download Transcript',
+    id: 'components.Transcripts.download',
+  },
+});
+
+/*
+ * Component. Displays the available choices for transcripts and plays a transcript when the user selects it.
+ */
+export const Transcripts = () => {
+  const intl = useIntl();
+  const video = useCurrentVideo();
+  const timedTextTracks = useTimedTextTrack((state) =>
+    state.getTimedTextTracks(),
+  );
+
+  const { choices, getChoices } = useTimedTextTrackLanguageChoices(
+    (state) => state,
+  );
+
+  const transcripts = timedTextTracks
+    .filter((track) => track.is_ready_to_show)
+    .filter((track) =>
+      video.has_transcript === false && video.should_use_subtitle_as_transcript
+        ? timedTextMode.SUBTITLE === track.mode
+        : timedTextMode.TRANSCRIPT === track.mode,
+    ) as TimedTextTranscript[];
+
+  const options = transcripts.map((transcript) => {
+    const language =
+      choices &&
+      choices.find(
+        (languageChoice) => languageChoice.value === transcript.language,
+      );
+    return {
+      label: language ? language.label : transcript.language,
+      value: transcript.id,
+    };
+  });
+
+  const [selectedOption, setSelectedOption] = useState({
+    label: intl.formatMessage(messages.transcriptSelectPlaceholder),
+    value: '',
+  });
+
+  const [selectedTranscript, setSelectedTranscript] = useState<{
+    language: string;
+    transcript: Nullable<TimedTextTranscript>;
+  }>({
+    language: '',
+    transcript: null,
+  });
+
+  // transcripts are not passed as a dependency because we only want them to load once
+  const onSelectChange = useCallback(
+    (option: { label: string; value: string }) => {
+      if (option.value === '') {
+        return setSelectedTranscript({
+          language: '',
+          transcript: null,
+        });
+      }
+      const transcript = transcripts.find((ts) => ts.id === option.value);
+      if (transcript) {
+        setSelectedTranscript({
+          language: option.label,
+          transcript,
+        });
+      }
+    },
+    [],
```

Comment:
if timedTextTracks ressources are added or removed after your component is mounted you will have a potential bug, transcripts will not be synch.